tests/unit-tests/gconstruct/test_remap_result.py

```diff
@@ -535,11 +537,57 @@ def test_parse_config():
         assert predict_dir is None
         assert emb_dir is None
 
-if __name__ == '__main__':
-    test_parse_config()
+@pytest.mark.parametrize("num_rows", [1000, 10001])
+def test_idmap_save_no_duplicates(num_rows, monkeypatch):
+    # Mock GIB_BYTES to 1kib to force multiple partitions
+    monkeypatch.setattr("graphstorm.gconstruct.id_map.GIB_BYTES", 1024)
 
-    test_write_data_csv_file()
-    test_write_data_parquet_file()
-    test__get_file_range()
-    test_worker_remap_edge_pred()
-    test_worker_remap_node_data("pred")
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # Create a table with known IDs to make verification easier
+        # string ids will be sequential id_<idx>
+        ids = [f"id_{i}" for i in range(num_rows)]
+
+        # Create the IdMap
+        id_map = IdMap(np.array(ids))
+
+        # Save the id map - will create multiple partitions due to small GIB_BYTES
+        mapping_prefix = os.path.join(tmpdirname, "raw_id_mappings")
+        id_map.save(mapping_prefix)
+
```

Comment:
Do we know how many files exactly will be saved? If so, we might check the number of files here.

Reply:
We can get an approximate number if we determine the byte size of each row or the total byte size of the mapping table in memory